### PR TITLE
Added large payload note, new API parameter

### PR DESCRIPTION
**Pull request by hannanelson**
[Anypoint MQ REST APIs _ MuleSoft Documentation.pdf](https://github.com/mulesoft/docs-mq/files/6281272/Anypoint.MQ.REST.APIs._.MuleSoft.Documentation.pdf)
[Configure Queues _ MuleSoft Documentation.pdf](https://github.com/mulesoft/docs-mq/files/6281273/Configure.Queues._.MuleSoft.Documentation.pdf)
[Tutorial_ Create a Queue and Send Messages to It _ MuleSoft Documentation.pdf](https://github.com/mulesoft/docs-mq/files/6281274/Tutorial_.Create.a.Queue.and.Send.Messages.to.It._.MuleSoft.Documentation.pdf)

### CODE REVIEW

File: modules/ROOT/pages/_partials/mq-big-payload.adoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// MQ Big Payload
+// tag::mqBigPayload[]
+Payloads larger than 1 MB don't appear in the Message Browser.
+You can retrieve them with the Anypoint Broker API.
```
Comment (danieldatri) [inline modules/ROOT/pages/_partials/mq-big-payload.adoc:4]: Wouldn't be better if we attach the screenshot with the warning message the user can get on the UI for this case?

**Review by danieldatri — CHANGES_REQUESTED** [review]
Hi Hanna,
take a look of the comment I added. Other than that the rest seems to be fine to me.

File: modules/ROOT/pages/_partials/mq-big-payload.adoc (hunk 1 above)
Comment (hannanelson) [reply modules/ROOT/pages/_partials/mq-big-payload.adoc:4]: @danieldatri - the error message seems pretty self-explanatory, so I don't think we need to have a screenshot of it.

File: modules/ROOT/pages/mq-apis.adoc [hunk 2]
```diff
@@ -136,11 +133,58 @@ The command returns output similar to:
...
+==== Example: Retrieve a Message
+
+The following curl command gets a message:
+
+[source,console,linenums]
+----
+curl -X GET -H "Authorization: bearer 42544254-4254-4254-4254-425442544254" \
+-H "Cache-Control: no-cache" \
+-H "Postman-Token: 42544254-4254-4254-4254-425442544254" \
+"https://mq-us-east-1.anypoint.mulesoft.com/api/v1/organizations/42544254-4254-4254-4254-425442544254/environments/42544254-4254-4254-4254-425442544254/destinations/postmanQueue/messages?pollingTime=10000&batchSize=1&lockTtl=10000"
```
Comment (kimayap03) [inline modules/ROOT/pages/mq-apis.adoc:176]: Can you add the `payloadVisibility` parameter n the example query as well?

Comment (kimayap03) [reply modules/ROOT/pages/mq-apis.adoc:176]: We should also report `payloadHidden=true` in the response when message is hidden

Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc:176]: @kimayap03 - updated with your comments.

File: modules/ROOT/pages/mq-apis.adoc [hunk 3]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+This parameter is useful when retrieving large payloads.
+
+The `payloadVisibility` parameter takes the following options: 
+
+* `full`: Shows the message payload in the response+
+This option is the default.
+* `none`: Hides the message payload in the response
+* `conditional`: Hides the message payload if the payload size is larger than 1 MB. 
+
+If the value of `payloadVisibility` is `none`, the response includes `payloadHidden=true` in the header.
```
Comment (kimayap03) [inline modules/ROOT/pages/mq-apis.adoc]: @hannanelson though this statement is true, it doesnt offer complete information. 
Can we rephrase it to say something like If payload is hidden as a result of param value `none` or `conditional`, response will include`payloadHidden=true` in the message header.

File: modules/ROOT/pages/mq-apis.adoc [hunk 4]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+
+The following curl command gets a message and displays the payload only if the size is smaller than 1 MB:
```
Comment (kimayap03) [inline modules/ROOT/pages/mq-apis.adoc]: this gives illusion that we will not retrieve the message if payload is < 1MB, where as curl command response will still `get` the message

File: modules/ROOT/pages/mq-apis.adoc (hunk 3 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc]: Clarified.

**Review by kimayap03 — DISMISSED** [review]
LGTM!

File: modules/ROOT/pages/mq-apis.adoc [hunk 5]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
 {"access_token":"<token>","simple_client":{"envId":"ENVIRONMENT_ID","orgId":"ORGANIZATION_ID"},"token_type":"bearer"}
 ----
 
-https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/anypoint-mq-broker/[Broker API Portal] - Publish, consume, ack, and nack messages between queues and message exchanges.
 
-In the Broker API portal, click *Download* to download the API as a RAML, Mule 3 zipped connector file, or as a Mule 4 connector JAR file.
+[[excoliuscu]]
+=== Use the Anypoint MQ Broker API
+
+These examples use the Anypoint MQ Broker API to publish and consume messages and test Anypoint MQ from the command line using curl.
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
These examples use the Anypoint MQ Broker API to publish and consume messages and to test Anypoint MQ from the command line using curl.
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 6]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+
+You must substitute the placeholder string, `42544254-4254-4254-4254-425442544254` for the proper values for authorization and token strings. These examples use Postman to access the queue.
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
You must substitute the placeholder string, `42544254-4254-4254-4254-425442544254` with the proper values for authorization and token strings. These examples use Postman to access the queue.
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 7]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+
+See the xref:mq-faq.adoc#regions[MQ FAQ] for the regions where Anypoint MQ and FIFO are available.
+
+==== Example: Publish a Message
+
+The following *curl* command publishes a message:
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
The following `curl` command publishes a message:
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 8]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+  },
+
+  "body": "This is a message payload"
+}
+' "https://mq-us-east-1.anypoint.mulesoft.com/api/v1/organizations/42544254-4254-4254-4254-425442544254/environments/42544254-4254-4254-4254-425442544254/destinations/postmanExchange/messages/552"
+----
+
+==== Example: Retrieve a Message
+
+The following curl command gets a message:
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
The following `curl` command gets a message:
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 9]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+"https://mq-us-east-1.anypoint.mulesoft.com/api/v1/organizations/42544254-4254-4254-4254-425442544254/environments/42544254-4254-4254-4254-425442544254/destinations/postmanQueue/messages?pollingTime=10000&batchSize=1&lockTtl=10000"
+----
+
+
+You can control whether to show the message payload in the response using the `payloadVisibility` query parameter.
+This parameter is useful when retrieving large payloads.
+
+The `payloadVisibility` parameter takes the following options: 
+
+* `full`: Shows the message payload in the response+
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
* `full`: Shows the message payload in the response. +
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 10]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+This option is the default.
+* `none`: Hides the message payload in the response
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
* `none`: Hides the message payload in the response.
```

**Review by prafulrana — DISMISSED** [review]

File: modules/ROOT/pages/mq-apis.adoc [hunk 11]
```diff
@@ -136,11 +133,72 @@ The command returns output similar to:
...
+* `conditional`: Hides the message payload if the payload size is larger than 1 MB. 
+
+If the payload is hidden because `payloadVisibility` is set to `none` or `conditional`, the response includes `payloadHidden=true` in the message header.
+
+The following curl command gets a message and payload, displaying the payload only if the payload size is smaller than 1 MB:
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc]:
```suggestion
The following `curl` command gets a message and payload, displaying the payload only if the payload size is smaller than 1 MB:
```

File: modules/ROOT/pages/mq-apis.adoc [hunk 12]
```diff
@@ -798,6 +815,16 @@ Ensure that the Console messages end with the `DEPLOYED` value.
 
 After you set up an HTTP Listener in Studio, browse to the address `0.0.0.0:8081`. The browser displays the message sent by Postman, which the Anypoint MQ connector received, and the HTTP connector sent to the browser.
 
+== Download Anypoint MQ APIs from Exchange
+
+To download the Anypoint MQ APIs from Exchange, click *Download* and select the format to download: RAML, OAS, or as a connector file for Mule 3 or Mule 4.
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc:820]: minor: "as a" sounds kind of odd, but I don't have a good alternative.

File: modules/ROOT/pages/mq-apis.adoc [hunk 13]
```diff
@@ -798,6 +815,16 @@ Ensure that the Console messages end with the `DEPLOYED` value.
...
+
+.The arrow shows the *Download* button in Exchange.
```
Comment (dukesphere) [inline modules/ROOT/pages/mq-apis.adoc:822]:
```suggestion
.The arrow shows the *Download* button in Exchange:
```

Comment (dukesphere) [reply modules/ROOT/pages/mq-apis.adoc:822]: Your call. I like colons on these because of the way the captions look in HTML. It's not always clear that they are captions for something that comes next. Same goes for the next one(s).

**Review by dukesphere — COMMENTED** [review]
some suggestions

File: modules/ROOT/pages/mq-apis.adoc (hunk 13 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc:822]: I don't put the colon on captions because I put them in the intro sentence (per Kim style).

File: modules/ROOT/pages/mq-apis.adoc (hunk 12 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc:820]: Stole that sentence from a Kim-approved version in OSv2...

File: modules/ROOT/pages/mq-apis.adoc (hunk 10 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc]: Not a complete sentence, so leaving.

**Review by dukesphere — DISMISSED** [review]

File: modules/ROOT/pages/mq-apis.adoc (hunk 10 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-apis.adoc]: nm, implemented.

**Review by dukesphere — DISMISSED** [review]

**Review by danieldatri — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/mq-faq.adoc [hunk 14]
```diff
@@ -82,6 +82,11 @@ Anypoint MQ limits messages to 10 MB.
 
 If a larger size is required, you can split the payload into multiple messages, or implement a claim check style service in which you store the payload in a file system or blob storage, and then put a pointer to the payload in the message so it can be downloaded later. However, you must manage access control to the blob storage for both the sender and receiver. You also need to consider that messages can be received unordered unless you use a FIFO queue.
 
+The *Message Browser* page displays the payload only if the payload size is smaller that 1 MB.
```
Comment (danieldatri) [inline modules/ROOT/pages/mq-faq.adoc]: Shouldn't be say **smaller than** instead of **smaller that**?
If I'm wrong I was wrong my hole life thinking  that was the correct wait to say it.  :)

**Review by danieldatri — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/mq-faq.adoc (hunk 14 above)
Comment (hannanelson) [reply modules/ROOT/pages/mq-faq.adoc]: You're RIGHT. Typo!

Comment (hannanelson) [reply modules/ROOT/pages/mq-faq.adoc]: Fixed in the RN PR too.

**Review by danieldatri — APPROVED** [review]
LGTM

**Review by dukesphere — APPROVED** [review]
